Apex7000_BillValidator/Serial/BaseDataValidator.cs

```diff
@@ -0,0 +1,73 @@
+﻿ namespace PyramidNETRS232.Serial
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    
+    /// <summary>
+    /// Base class for defining serial data validation
+    /// </summary>
+    public abstract class BaseDataValidator
```

Comment:
Can these validators be internal?

Reply:
The classes that inherit BaseDataValidator can be.

BaseDataValidator was left non-internal since a BaseDataValidator is used as a property in ProtocolViolationEventArgs.

Reply:
Oh okay, I didn't see that it was exposed as an event arg.